### PR TITLE
EPMLSTRCMW-261 Feat: Changed ep in RunController api to be more restful

### PR DESCRIPTION
**Pull request by DruzhininPavel**
**DESCRIPTION**:
change endpoints in RunController to the REST-full

**CHANGES**:
Deleted DeleteRequest for Run

### CODE REVIEW

File: services/src/main/scala/cromwell/pipeline/service/RunService.scala [hunk 1]
```diff
@@ -1,31 +1,32 @@
...
 import cromwell.pipeline.datastorage.dto._
 import cromwell.pipeline.model.wrapper.{ RunId, UserId }
 
+import java.time.Instant
 import scala.concurrent.{ ExecutionContext, Future }
 
 trait RunService {
 
-  def addRun(runCreateRequest: RunCreateRequest): Future[RunId]
+  def addRun(runCreateRequest: RunCreateRequest, userId: UserId): Future[RunId]
```
Comment (Nikolai-Isaev) [inline services/src/main/scala/cromwell/pipeline/service/RunService.scala:12]: Why do we have to pass `userId` if `runCreateRequest` contains one? What am I missing? :)

Comment (DruzhininPavel) [reply services/src/main/scala/cromwell/pipeline/service/RunService.scala:12]: Yep. You are right! I`l fix it.

**Review by Nikolai-Isaev — APPROVED** [review]
Lgtm

**Review by gominasian — APPROVED** [review]
Well done!

File: controllers/src/main/scala/cromwell/pipeline/controller/utils/PathMatchers.scala [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+object PathMatchers {
+  val ProjectId: PathMatcher1[dto.ProjectId] = Segment.map(dto.ProjectId(_))
+  val Path: PathMatcher1[Path] = Segment.map(Paths.get(_))
+  val RunId: PathMatcher1[wrapper.RunId] = Segment.flatMap {
+    wrapper.RunId.fromString(_) match {
+      case Validated.Valid(runId) => Some(runId)
+      case _                      => None
+    }
+  }
```
Comment (myazinn) [inline controllers/src/main/scala/cromwell/pipeline/controller/utils/PathMatchers.scala]:
```suggestion
  val RunId: PathMatcher1[wrapper.RunId] = Segment.flatMap(wrapper.RunId.fromString(_).toOption)
```
Please always check whether the required method exists before reimplementing it yourself.

File: controllers/src/test/scala/cromwell/pipeline/controller/RunControllerTest.scala [hunk 3]
```diff
@@ -49,47 +49,44 @@ class RunControllerTest
         when(runService.getRunByIdAndUser(runId, accessToken.userId))
           .thenReturn(Future.failed(new RuntimeException("something went wrong")))
 
-        Get("/runs?run_id=" + runId.unwrap) ~> runController.route(accessToken) ~> check {
+        Get("/runs/" + runId.unwrap) ~> runController.route(accessToken) ~> check {
           status shouldBe StatusCodes.InternalServerError
         }
       }
 
       "returns run entity if the run id is not valid" taggedAs Controller in {
```
Comment (myazinn) [inline controllers/src/test/scala/cromwell/pipeline/controller/RunControllerTest.scala]:
```suggestion
      "returns NotFound if the run id is not valid" taggedAs Controller in {
```
This is your mistake, but please fix it while we are here

File: controllers/src/test/scala/cromwell/pipeline/controller/RunControllerTest.scala [hunk 4]
```diff
@@ -49,47 +49,44 @@ class RunControllerTest
...
         }
       }
       "returns server error if run delete was failed" taggedAs Controller in {
         val runId = TestRunUtils.getDummyRunId
-        val deleteRequest = RunDeleteRequest(runId)
         when(runService.deleteRunById(runId, accessToken.userId))
           .thenReturn(Future.failed(new RuntimeException("something went wrong")))
 
-        Delete("/runs", deleteRequest) ~> runController.route(accessToken) ~> check {
+        Delete("/runs/" + runId.unwrap) ~> runController.route(accessToken) ~> check {
```
Comment (myazinn) [inline controllers/src/test/scala/cromwell/pipeline/controller/RunControllerTest.scala]: Actually we don't need `.unwrap` everywhere. `s"/runs/$runId"` should work just fine

File: model/src/main/scala/cromwell/pipeline/model/wrapper/RunId.scala [hunk 5]
```diff
@@ -19,6 +19,11 @@ object RunId extends Wrapped.Companion {
     value,
     NonEmptyChain.one("Invalid RunId")
   )
+  def fromString(value: String): ValidationResult[Wrapper] =
```
Comment (myazinn) [inline model/src/main/scala/cromwell/pipeline/model/wrapper/RunId.scala]: Why do we need this method? Can't we just use `.from` ?

File: services/src/main/scala/cromwell/pipeline/service/RunService.scala [hunk 6]
```diff
@@ -1,43 +1,45 @@
...
           timeStart = Instant.now(),
-          userId = runCreateRequest.userId,
+          userId = userId,
           results = runCreateRequest.results
         )
-
-        runRepository.addRun(newRun)
+        projectService
+          .getUserProjectById(runCreateRequest.projectId, userId: UserId)
+          .flatMap(_ => runRepository.addRun(newRun))
```
Comment (myazinn) [inline services/src/main/scala/cromwell/pipeline/service/RunService.scala]: Although it is a reasonable change, let's make it as a separate task. Please create a ticket for it.

**Review by myazinn — CHANGES_REQUESTED** [review]

File: controllers/src/main/scala/cromwell/pipeline/controller/utils/PathMatchers.scala (hunk 2 above)
Comment (DruzhininPavel) [reply controllers/src/main/scala/cromwell/pipeline/controller/utils/PathMatchers.scala]: Opps, sorry. I forgot about that. I`ll fix it.

File: services/src/main/scala/cromwell/pipeline/service/RunService.scala (hunk 6 above)
Comment (DruzhininPavel) [reply services/src/main/scala/cromwell/pipeline/service/RunService.scala]: Ok. I`ll rollback this changes and create new task.

File: model/src/main/scala/cromwell/pipeline/model/wrapper/RunId.scala (hunk 5 above)
Comment (DruzhininPavel) [reply model/src/main/scala/cromwell/pipeline/model/wrapper/RunId.scala]: Yep. You are right! I`ll change it.

File: controllers/src/test/scala/cromwell/pipeline/controller/RunControllerTest.scala (hunk 4 above)
Comment (DruzhininPavel) [reply controllers/src/test/scala/cromwell/pipeline/controller/RunControllerTest.scala]: Hmmm. It`s really works. Thanks.

File: controllers/src/main/scala/cromwell/pipeline/controller/utils/PathMatchers.scala (hunk 2 above)
Comment (DruzhininPavel) [reply controllers/src/main/scala/cromwell/pipeline/controller/utils/PathMatchers.scala]: Done.

File: controllers/src/test/scala/cromwell/pipeline/controller/RunControllerTest.scala (hunk 3 above)
Comment (DruzhininPavel) [reply controllers/src/test/scala/cromwell/pipeline/controller/RunControllerTest.scala]: Yep.My fault. Done.

File: controllers/src/test/scala/cromwell/pipeline/controller/RunControllerTest.scala (hunk 4 above)
Comment (DruzhininPavel) [reply controllers/src/test/scala/cromwell/pipeline/controller/RunControllerTest.scala]: Changed.

File: model/src/main/scala/cromwell/pipeline/model/wrapper/RunId.scala (hunk 5 above)
Comment (DruzhininPavel) [reply model/src/main/scala/cromwell/pipeline/model/wrapper/RunId.scala]: Done.

File: services/src/main/scala/cromwell/pipeline/service/RunService.scala (hunk 6 above)
Comment (DruzhininPavel) [reply services/src/main/scala/cromwell/pipeline/service/RunService.scala]: Done.

**Review by Dmitrii-Shevchenko — APPROVED** [review]
Didn't find any problem. Good work!

**Review by myazinn — APPROVED** [review]

**Pull request merged**